hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/HoodieMetadataWriters.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.client;
+
+import org.apache.hudi.common.data.HoodieData;
+import org.apache.hudi.common.model.HoodieCommitMetadata;
+import org.apache.hudi.common.model.HoodieWriteStat;
+import org.apache.hudi.common.table.HoodieTableVersion;
+import org.apache.hudi.common.util.Option;
+import org.apache.hudi.common.util.ValidationUtils;
+import org.apache.hudi.exception.HoodieException;
+import org.apache.hudi.metadata.HoodieTableMetadataWriter;
+import org.apache.hudi.table.HoodieTable;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Utilities for data table write client and table service client to write to metadata table.
+ */
+public class HoodieMetadataWriters {
```

Comment:
I am not convinced this is a good name @danny0405  

How about HoodieMetadataWriteDelegate or HoodieMetadataWriteCoordinator ? 
@the-other-tim-brown : you also had some suggestions on the naming right.

Reply:
`XXXs` usually is a naming convention for utilities of `XXX`, which works for most of the cases.

Reply:
I have not heard of this convention before. My suggestion was [StreamingMetadataWriteHandler](https://github.com/apache/hudi/pull/13402#discussion_r2145857643) - I am not too picky either way but if this class is solely for the streaming write support then it should at least be mentioned in the javadoc if not the class name.

Reply:
StreamingMetadataWriteHandler sounds good w/ me.